### PR TITLE
Fix metrics submission issue and handle errors

### LINKED ISSUE #1359: [BUG]: Configure bias metric form does not work on submission
Tracker: #1127 

When testing with the latest version of TrustyAI the "Configure bias metric form" doesn't work. When submitting valid data:

Screenshot 2023-06-12 at 11 08 19

The form disappears and reappears and no metric has been configured. The network tab of the browser console shows the following error:
```JSON
{
    "title": "Constraint Violation",
    "status": 400,
    "violations":
    [
        {
            "field": "createRequest.request",
            "message": "Invalid type for outcome. Got '\"0\"', expected object compatible with 'INT64'"
        },
        {
            "field": "createRequest.request",
            "message": "The supplied metric request details are not valid."
        }
    ]
}
```

The problem is that we need to send numbers, booleans or strings as their correct JSON type, not just as a string. We will have to parse what the user inputs so we can send it as the correct value. If they've entered the string `"true"` send boolean `true`, or if they've entered `"1.0"` or `"1"` send it as a number, `"foobarbaz"` should obviously stay as a string.

Essentially these values here will need to be retyped as `string | number | boolean` : https://github.com/opendatahub-io/odh-dashboard/blob/4aa8675dc41be45b3c3bff9d2e10a94966cbe43f/frontend/src/api/trustyai/rawTypes.ts#L30-L32

Also - if the form could use the standard error object TrustyAI provides above to provide the user with a list of errors, that would provide a better UX.

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
Closes #1359 

## Description

1. Change the type of `privilegedAttribute`, `unprivilegedAttribute` and `favorableOutcome` to meet the trusty AI server requirements for data types.
2. Create error handlers to handle the 4xx error that comes from the trusty AI server. For example, when you input some invalid value, trusty AI server will give a 400 Bad Request error back, we can show it in the creation modal now.

Screenshot 2023-06-13 at 3 09 52 PM

3. Fix the problem that users cannot successfully delete SPD metrics.
4. Check the session storage every time entering the bias charts and update them. Because when users delete one configuration, we didn't update the session storage for the chart selections, we need to reflect all the changes that the user made.

## How Has This Been Tested?

To test description 1:
Try to create/duplicate a configuration, using the valid attributes to see if you could successfully create one

To test description 2:
Try to create/duplicate a configuration, using the invalid attributes like string for the `privilegedAttribute` field to see if you can get the error alert after hitting the submit button

To test description 3:
Try to delete an SPD type configuration (You cannot delete that before because the context used the wrong delete function)

To test description 4:
Select to show a metric to display -> Go to the configuration page -> Delete that metric -> Go back to view the metrics chart -> Make sure the selection is removed and the chart is also removed

## Test Impact

TBD

## Request review criteria:

- [x] The commits have meaningful messages (squashes happen on merge by the bot).
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

### CODE REVIEW

File: frontend/src/api/trustyai/custom.ts [hunk 1]
```diff
@@ -18,17 +19,19 @@ export const getSpdRequests =
...
+      proxyDELETE(
+        hostPath,
+        '/metrics/spd/request',
+        { requestId: id },
+        {},
+        { parseJSON: false, ...opts },
+      ),
     );
 
 export const getDirRequests =
```
Comment (alexcreasy) [inline frontend/src/api/trustyai/custom.ts:37]: Do we not need to handle the error for get requests also?

File: frontend/src/concepts/explainability/useTrustyAPIState.ts [hunk 2]
```diff
@@ -22,7 +23,7 @@ const useTrustyAPIState = (
       createDirRequest: createDirRequest(path),
       createSpdRequest: createSpdRequest(path),
       deleteDirRequest: deleteDirRequest(path),
-      deleteSpdRequest: deleteDirRequest(path),
+      deleteSpdRequest: deleteSpdRequest(path),
```
Comment (alexcreasy) [inline frontend/src/concepts/explainability/useTrustyAPIState.ts:26]: Good catch :/

**Review by alexcreasy — COMMENTED** [review]
Just one question in there, otherwise looks good, works well and we can merge.

File: frontend/src/api/trustyai/custom.ts (hunk 1 above)
Comment (DaoDaoNoCode) [reply frontend/src/api/trustyai/custom.ts:37]: Hmm, I haven't found any get errors yet, but we can definitely handle them all.

**Comment by alexcreasy** [comment]
lgtm

**Pull request merged**